exchanges/binance/binance_types.go

```diff
@@ -200,6 +200,18 @@ type HistoricalTrade struct {
 	IsBestMatch  bool    `json:"isBestMatch"`
 }
 
+// AggregatedTradeRequestParams holds request params
+type AggregatedTradeRequestParams struct {
+	Symbol currency.Pair // Required field; example LTCBTC, BTCUSDT
```

Comment:
hmm.. really? I think using the currency.Pair keeps the "type safety", the same reason there's `time.Time` in the start and end times, not just `int64` or a `string` (everything gets converted to string for the url parameter in the end, but as the caller I shouldn't care if the timestamp is encoded in milliseconds or seconds; or if the currency pair is encoded with '-' as the separator or empty string) - i.e. normally you want to push the "safe types" all the way down so the API would be easier to understand and you can do the conversion from the "stronger type" to "weaker one" in a single place only when you need it. 
I do recognise that might be an extra dependency the callers might need, but the dependency is already there for the package and I believe it is small price to pay for extra safety.
If you still disagree and think that the exchange API should be as generic as possible, I'm OK to change it

Reply:
Across the codebase we've made the exchange API as generic/native as possible (for people who don't care about the GCT built-ins) and have instead made the wrappers utilise the GCT built-ins and convert them to the generic types. That way, a person can choose between whether they want the GCT wrapped functionality or just the generic API implementation. I do see your point though in pushing the safe types all the way to the API level and it could be addressed in a separate PR as it would be a large undertaking.

Reply:
> for people who don't care about the GCT built-ins
I still fail to understand the use case... if I just wanted a way to send requests to exchange API and handle the serialisation on my own, I would use curl or dynamically typed JavaScript or Python, not a golang library - this is exactly the added value that I expect from the library - to make using the exchange API easier. This includes not being forced to think about low level serialisation issues, but in core concepts.